src/pages/workspaces/workspace/notebooks/NotebookLauncher.js

```diff
@@ -32,37 +91,300 @@ const NotebookLauncher = _.flow(
     title: _.get('notebookName'),
     showTabBar: false
   })
-)(({ queryParams = {}, notebookName, workspace, workspace: { accessLevel, canCompute }, cluster, refreshClusters, onRequesterPaysError }, ref) => {
-  return (Utils.canWrite(accessLevel) && canCompute && !queryParams['read-only']) ?
-    h(NotebookEditor, { notebookName, workspace, cluster, refreshClusters }) :
-    h(Fragment, [
-      h(ReadOnlyMessage, { notebookName, workspace }),
-      h(NotebookPreviewFrame, { notebookName, workspace, onRequesterPaysError })
+)(
+  ({ queryParams = {}, notebookName, workspace, workspace: { workspace: { namespace, name }, accessLevel, canCompute }, cluster, refreshClusters, onRequesterPaysError }, ref) => {
+    const [createOpen, setCreateOpen] = useState(false)
+    // Status note: undefined means still loading, null means no cluster
+    const clusterStatus = cluster && cluster.status
+    const [busy, setBusy] = useState()
+    const mode = queryParams['mode']
+
+    return h(Fragment, [
+      (Utils.canWrite(accessLevel) && canCompute && !!mode && clusterStatus === 'Running') ?
+        h(cluster.labels.welderInstallFailed ? WelderDisabledNotebookEditorFrame : NotebookEditorFrame, { key: cluster.clusterName, workspace, cluster, notebookName, mode }) :
+        h(Fragment, [
+          h(PreviewHeader, { queryParams, cluster, refreshClusters, notebookName, workspace, readOnlyAccess: !(Utils.canWrite(accessLevel) && canCompute), onCreateCluster: () => setCreateOpen(true) }),
+          h(NotebookPreviewFrame, { notebookName, workspace })
+        ]),
+      mode && h(ClusterKicker, { cluster, refreshClusters, onNullCluster: () => { setCreateOpen(true) } }),
+      mode && h(ClusterStatusMonitor, { cluster, onClusterStoppedRunning: () => { chooseMode(undefined) } }),
+      h(NewClusterModal, {
+        isOpen: createOpen,
+        namespace, currentCluster: cluster,
+        onDismiss: () => {
+          chooseMode(undefined)
+          setCreateOpen(false)
+        },
+        onSuccess: withErrorReporting('Error creating cluster', async promise => {
+          setCreateOpen(false)
+          setBusy(true)
+          await promise
+          await refreshClusters()
+          setBusy(false)
+        })
+      }),
+      busy && spinnerOverlay
     ])
-})
+  })
 
-const ReadOnlyMessage = ({ notebookName, workspace, workspace: { canCompute, workspace: { namespace, name } } }) => {
-  const [copying, setCopying] = useState(false)
+const FileInUseModal = ({ onDismiss, onCopy, onPlayground, namespace, name, bucketName, lockedBy, canShare }) => {
+  const [lockedByEmail, setLockedByEmail] = useState(null)
```

Comment:
Implicit default is undefined, which is falsey.
```suggestion
  const [lockedByEmail, setLockedByEmail] = useState()
```